zcnbridge/transaction/functions.go

```diff
@@ -1,36 +1,29 @@
 package transaction
 
-// ZCNSC smart contract functions wrappers. Partially covered. At this stage not all required
+// ZCNSC smart contract functions wrappers
 
 import (
 	"context"
-	"encoding/json"
 
-	"github.com/0chain/gosdk/core/transaction"
 	"github.com/0chain/gosdk/zcncore"
 )
 
-func (t *Transaction) AddAuthorizer(ctx context.Context, input *zcncore.AddAuthorizerPayload) error {
+func AddAuthorizer(ctx context.Context, input *zcncore.AddAuthorizerPayload) (*Transaction, error) {
 	t, err := NewTransactionEntity()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	buffer, err := json.Marshal(input)
+	err = t.scheme.ZCNSCAddAuthorizer(input)
 	if err != nil {
-		return err
+		return t, err
 	}
 
-	_, err = t.ExecuteSmartContract(
-		ctx,
-		zcncore.ZCNSCSmartContractAddress,
-		transaction.ZCNSC_ADD_AUTHORIZER,
-		string(buffer),
-		0,
-	)
+	err = t.callBack.waitCompleteCall(ctx)
+	t.Hash = t.scheme.Hash()
```

Comment:
is it really needed?

Reply:
Yes, required for further transaction verification

Reply:
but Transaction has 2 different hash fields, one in t.scheme and another in t. 
Why we didn't embed t.scheme in transaction, so we could remove hash field duplication and use something like this?
```go
func (t *Transaction) Hash() string {
    return t.scheme.Hash()
 
}
```